src/plugins/upf/upf_pfcp_api.c

```diff
@@ -871,14 +865,8 @@ static int handle_update_pdr(upf_session_t *sess, pfcp_update_pdr_t *update_pdr,
 			{
 				update->pdi.fields |= F_PDI_APPLICATION_ID;
 
-				pfcp_application_id_t *app_id = NULL;
 				pfcp_application_id_t *app_id_vector = NULL;
-		
-				vec_foreach(app_id, pdr->pdi.application_id)
-					{
-						vec_add1(app_id_vector, *app_id);
-					}
-
+				vec_add1(app_id_vector, pdr->pdi.application_id);
 				upf_add_multi_regex(app_id_vector, &update->dpi_db_id, 0);
```

Comment:
since there is now only one Application Id to add, the whole multi_regex logic could be simplified. The only called that can add multiple App Id is the CLI, and that can be changed.